code/modules/mob/living/living.dm

```diff
@@ -48,6 +48,8 @@
 				qdel(S)
 			else
 				S.be_replaced()
+	if(mind)
+		mind.current = null
```

Comment:
Is there a case where `current` can be something different? Like when ghosting or so?

Reply:
That would be mob/dead/observer, not a living type

Reply:
Aye, but will the original mob still have the mind attached to it? If that's the case then you'll unset the `current` while it's not  even the mob that's being deleted.

Reply:
You can re-enter your body multiple times, so yes

Reply:
How about `if(mind.current == src)` instead?

Reply:
Maybe? I'll try it tomorrow but I don't think it will have any actual effect on the implementation

Reply:
Yeah shouldn't. Just trying to address the "is their `mind.current` actually this src mob?" concern.

Reply:
> Yeah shouldn't. Just trying to address the "is their `mind.current` actually this src mob?" concern.

If that's not the case something horrible has gone wrong and it should be logged in some way then, I can change it tomorrow and add a stacktrace if it goes to shit

Reply:
```
	if(mind?.current == src)
		mind.current = null
```

Reply:
Applied, sorry this took awhile, just got home